src/ir/ty.rs

```diff
@@ -376,6 +412,8 @@ pub enum TypeKind {
     ///
     /// see tests/headers/typeref.hpp to see somewhere where this is a problem.
     UnresolvedTypeRef(clang::Type, Option<clang::Cursor>, /* parent_id */ Option<ItemId>),
+
+    /// An indirection to another type.
```

Comment:
It'd be nice to mention that we use them, at least:

 * To avoid parsing types that are forward declared too early (pointing to `UnresolvedTypeRef`).
 * To do type replacement without iterating the whole type graph.